lib/std/compress/tar/test_decompress.zig

```diff
@@ -0,0 +1,37 @@
+const std = @import("std");
+const mem = std.mem;
+const tar = std.tar;
+const testing = std.testing;
+
+test "tar decompress testdata" {
```

Comment:
The testdata/ folder is currently at 136K. I think each of the tar files should be gzipped.  And then this test, and maybe reader_test.zig, should use `@embedFile` rather than iterating the testdata/ directory.  Does that seem bettter? 

Thats whats being done in [the xz tests](https://github.com/ziglang/zig/blob/master/lib/std/compress/xz/test.zig#L21>